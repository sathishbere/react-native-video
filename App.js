import React, { useEffect, useState } from 'react';
import { Text, View, Button, Platform } from 'react-native';
import ReactPlayer from 'react-player';


const App = () => {
const [names, setNames] = useState([]);
const [vid, setVid]=useState([{}]);

let url= {web:"http://localhost:8089/api/names", ios:"http://localhost:8080/getVideo",android:"http://localhost:8085/getVideo"};

  useEffect(() => {
    try{
    fetch(url[Platform.OS]).then(response=>response.json().then(data=>setNames(data)))
    }catch(error){
      console.error(error)
    }
    }, []);

    const btnHandler=(id)=>{
      try{
      fetch(`http://localhost:8089/api/names/${id}`).then(response=>response.json().then(data=>setVid(data)))
      }catch(error){
        console.error(error)
      }
    }  

  return (

    <View style={{ flex: 1, padding: 24, justifyContent:'center', alignItems:'center', }}>      
    < Text testID={'Mynames'}>{names.map((el,ind)=><Button title="Start Video" key={ind} onPress={()=>btnHandler(ind)}/> )}</Text>
    <Text ><ReactPlayer playing url={vid.video} controls={true}
    onReady={()=>console.log("OnReady")}
    onStart={()=>console.log("OnStart")}
    onPause={()=>console.log("OnPause")}
    onEnded={()=>console.log("OnEnded")}
    onError={()=>console.log("OnError")}/></Text>
       
      </View>
  );
};

export default App;