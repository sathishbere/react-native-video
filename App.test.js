import React from 'react';
import {create, renderer} from 'react-test-renderer';
import App from './App';

jest.useFakeTimers();
jest.runAllTimers();

const snap=create(<App/>);
test('render properly', ()=>{
    expect(snap).toMatchSnapshot();
});

