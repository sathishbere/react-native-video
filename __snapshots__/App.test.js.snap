// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
    }
  }
>
  <Text
    testID="Mynames"
  >
    0
  </Text>
  <Text />
  <Text>
    <div
      style={
        Object {
          "height": "400",
          "width": "640px",
        }
      }
    />
  </Text>
</View>
`;
